ksql-engine/src/main/java/io/confluent/ksql/planner/plan/AggregateNode.java

```diff
@@ -393,16 +394,46 @@ private void collectAggregateArgExpressions(
           });
     }
 
-    List<Expression> getInternalExpressionList(final List<Expression> expressionList) {
-      return expressionList.stream()
-          .map(argExpression -> argExpression instanceof Literal
-                                ? argExpression
-                                : new QualifiedNameReference(
-                                    QualifiedName.of(getExpressionToInternalColumnNameMap()
-                                            .get(argExpression.toString()))))
+    List<Expression> getInternalExpressionList(final List<Expression> argExpressionList) {
+      return argExpressionList.stream()
+          .map(argExpression -> new QualifiedNameReference(
+              QualifiedName.of(getExpressionToInternalColumnNameMap()
+                  .get(argExpression.toString()))))
           .collect(Collectors.toList());
     }
 
+    /**
+     * Return the aggregate function arguments based on the internal expressions.
+     * Currently we support aggregate functions with at most two arguments where
+     * the second argument should be a literal.
+     * @param argExpressionList The list of parameters for the aggregate fuunction.
+     * @return The list of arguments based on the internal expressions for the aggregate function.
+     */
+    //TODO We need to generalize this implementation for aggregate functions with more than two args
+    List<Expression> getInternalArgsExpressionList(final List<Expression> argExpressionList) {
+      Objects.requireNonNull(argExpressionList, "Argument list for UDAF cannot be null.");
```

Comment:
Not need to requireNotNull here - it will through NPE anyway and in any case it can't currently be null.

Reply:
Removed the line.